tests/unit_tests/data_steward/gcloud/gcs/gcs_test.py

```diff
@@ -32,6 +34,27 @@ def setUp(self):
         self.bucket: str = 'foo_bucket'
         self.prefix: str = 'foo_prefix/'
         self.file_name: str = 'foo_file.csv'
+        self.hpo_id = 'fake_hpo_id'
+
+    def test_unset_bucket(self):
+        bucket_env_var = f'BUCKET_NAME_{self.hpo_id.upper()}'
+        expected_message = lambda bucket: f"Bucket '{bucket}' for hpo '{self.hpo_id}' is unset/empty"
+
+        # run without setting env var (unset env_var)
+        os.environ.pop(f"BUCKET_NAME_{self.hpo_id.upper()}", None)
```

Comment:
env var use has the potential to disrupt the namespace, especially if tests are run in parallel.  the dict `os.environ` can be patched:

```
@mock.patch.dict(os.environ, {'mock_key': 'mock_value'})
def test_mock_env(self):
    self.assertEqual(os.environ['mock_key'], 'mock_value')
```
[See here for other ideas.](https://adamj.eu/tech/2020/10/13/how-to-mock-environment-variables-with-pythons-unittest/)